src/libs/Network/processRequest.js

```diff
@@ -11,11 +32,43 @@ import * as NetworkEvents from './NetworkEvents';
  * @returns {Promise}
  */
 export default function processRequest(request) {
+    const persisted = lodashGet(request, 'data.persist', false);
     const finalParameters = enhanceParameters(request.command, request.data);
 
     // When the request goes past a certain amount of time we trigger a re-check of the connection
     const cancelRequestTimeoutTimer = NetworkEvents.startRecheckTimeoutTimer();
-    NetworkEvents.triggerRequestMade(request, finalParameters);
+    logRequestDetails(request, finalParameters);
     return HttpUtils.xhr(request.command, finalParameters, request.type, request.shouldUseSecure)
+        .then((response) => {
+            if (persisted) {
+                PersistedRequests.remove(request);
+            }
+            NetworkEvents.triggerResponse(request, response);
+            return response;
+        })
+        .catch((error) => {
+            if (error.message === CONST.ERROR.FAILED_TO_FETCH) {
+                // Throw when we get a "Failed to fetch" error so we can retry. Very common if a user is offline or experiencing an unlikely scenario like
```

Comment:
NAB: I wonder if it's better to move the retry logic also here

Reply:
Thanks! But I'm not gonna do this either. Combining it would be less clear because we are interacting with 2 different queues and would need to create a circular dependency since they both call `processRequest()`.